### PR TITLE
feat(interactions): add the app_permissions field

### LINKED ISSUE #720: Adding the `app_permissions` field to nextcord
### Summary

This is a request to add the new `app_permissions` field/property to nextcord

### What is the feature request for?

Application commands

### The Problem

Discord recently added a new `app_permissions` field. Which, this should be added to nextcord's interaction class. This would return the bot's resolved permissions in the channel.

### The Ideal Solution

To add the `app_permissions` attribute to interactions

### The Current Solution

_No response_

### Additional Context

The documentation pull request that was ~~instantly~~ merged: https://github.com/discord/discord-api-docs/pull/5131

The documentation/change-log: https://discord.com/developers/docs/change-log#calculated-permissions-in-interaction-payloads

The announcement:
image

### PR DESCRIPTION
**Pull request by EmreTech**
## Summary

This uses the newly added `app_permissions` field for the permissions property in Interactions. Closes #720.

## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/interactions.py [hunk 1]
```diff
@@ -204,23 +204,10 @@ def _from_data(self, data: InteractionPayload):
...
-                pass
-            else:
-                self.user = Member(state=self._state, guild=guild, data=member)  # type: ignore
-                self._permissions = int(member.get("permissions", 0))
-        else:
-            try:
-                self.user = User(state=self._state, data=data["user"])
-            except KeyError:
-                pass
+            self._app_permissions = int(data.get("app_permissions"))  # type: ignore
```
Comment (DenverCoder1) [inline nextcord/interactions.py:210]: I think self.guild check can be removed as well and just have
`data.get("app_permissions", 0)`

The property could also stay as `_permissions` I think to reduce the PR noise (fewer changes)

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/interactions.py [hunk 2]
```diff
@@ -204,23 +204,7 @@ def _from_data(self, data: InteractionPayload):
             self.message = None
 
         self.user: Optional[Union[User, Member]] = None
```
[the remark this answers is not in the document]
Comment (ooliver1) [reply nextcord/interactions.py:206]:
```diff
         self.user: Optional[Union[User, Member]] = None
-        self._permissions: int = 0
-
-        # TODO: there's a potential data loss here
-        if self.guild_id:
-            guild = self.guild or Object(id=self.guild_id)
-            try:
-                member = data["member"]
-            except KeyError:
-                pass
-            else:
-                self.user = Member(state=self._state, guild=guild, data=member)  # type: ignore
-                self._permissions = int(member.get("permissions", 0))
-        else:
-            try:
-                self.user = User(state=self._state, data=data["user"])
-            except KeyError:
-                pass
+        self._permissions: int = int(data.get("app_permissions", 0))
```

[the remark this answers is not in the document]
Comment (EmreTech) [reply nextcord/interactions.py:206]: Ah right

**Comment by EmreTech** [comment]
> So it looks like `permissions` is permissions for the _user_ and `app_permissions` is permissions for the _bot_, so `permissions` probably shouldn't be touched at all.

Oh I thought `permissions` was for the bot
My bad

**Review by DenverCoder1 — APPROVED** [review]
Looks good, tested.

File: nextcord/interactions.py [hunk 3]
```diff
@@ -272,12 +274,20 @@ def channel(self) -> Optional[InteractionChannel]:
...
-        """:class:`Permissions`: The resolved permissions of the member in the channel, including overwrites.
+        """:class:`Permissions`: The resolved permissions of the member in the channel, including overwrites
 
         In a non-guild context where this doesn't apply, an empty permissions object is returned.
         """
         return Permissions(self._permissions)
 
+    @property
+    def app_permissions(self) -> Permissions:
+        """:class:`Permissions` The resolved permissions of the bot in the channel, including overwrites
```
Comment (DenverCoder1) [inline nextcord/interactions.py:285]:
```suggestion
        """:class:`Permissions`: The resolved permissions of the bot in the channel, including overwrites
```

File: nextcord/interactions.py [hunk 4]
```diff
@@ -272,12 +274,20 @@ def channel(self) -> Optional[InteractionChannel]:
 
     @property
     def permissions(self) -> Permissions:
-        """:class:`Permissions`: The resolved permissions of the member in the channel, including overwrites.
+        """:class:`Permissions`: The resolved permissions of the member in the channel, including overwrites
```
Comment (ooliver1) [inline nextcord/interactions.py:277]: confused as to why the dot was removed, its in almost every other docstring

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Review by ooliver1 — APPROVED** [review]

**Comment by DenverCoder1** [comment]
What happened to the colon after Permissions?

**Pull request merged**